src/WebJobs.Script.WebHost/Metrics/FlexConsumptionMetricsPublisher.cs

```diff
@@ -105,9 +106,12 @@ internal async Task OnPublishMetrics(DateTime now)
                     }
                 }
 
-                if (FunctionExecutionCount == 0 && FunctionExecutionTimeMS == 0 && !IsAlwaysReady && !_metricsProvider.HasMetrics())
+                bool hasActivity = FunctionExecutionCount > 0 || FunctionExecutionTimeMS > 0 || _metricsProvider.HasMetrics();
+                bool shouldForcePublish = (now - _lastPublishTime) >= TimeSpan.FromMilliseconds(_options.KeepAliveIntervalMS);
```

Comment:
Rather than just force emitting empty metrics, isn't there a new metric that you can add to HostMetricsProvider such that HasMetrics returns true on your required interval and this class behaves as it does now without any forced writing?

E.g. if your goal is to have some heartbeat metric, why not have HostMetricsProvider return one?

Reply:
That's a great suggestion, I like that idea a lot - will look into it